ocaml/typing/typecore.ml

```diff
@@ -8883,7 +8753,7 @@ and type_n_ary_function
     re
       { exp_desc =
           Texp_function
-            { params; body; region = region_locked; ret_sort;
+            { params; body; region = true; ret_sort;
```

Comment:
Now, `region` is always `true` for user-written functions and always `false` for functions that arise from coercing labeled functions. This seems suspicious to me -- what does coercing labeled functions have to do with regions? The reason I ask is: if we no longer need the `region` field, we can simplify translation somewhat.

Reply:
Yes you are right. I didn't do the simplification because it would involve lots of changes in `translcore` as well; but I guess it has to be done sometime anyway.

Reply:
Let’s do it in a separate PR.

Reply:
Maybe we can remvoe `region` from `Typedtree.Texp_function` in this PR, and do the changes in `translcore` in a separate PR? I agree with your that `Texp_function` still having `region` is annoying.

Reply:
So I did what I said (remove `region` from `Typedtree.Texp_function`, while try to not change `translcore.ml`).